packages/synapse-react-client/src/lib/containers/AccessRequirementListV2/AccessRequirementListV2.tsx

```diff
@@ -0,0 +1,347 @@
+  borderBottom: '1px solid',
+  borderColor: theme.palette.grey['200'],
+}))
+DialogSubsectionHeader.defaultProps = {
+  ...DialogSubsectionHeader.defaultProps,
+  variant: 'h4',
+}
+
+export const checkHasUnsupportedRequirement = (
+  accessRequirements: Array<AccessRequirement>,
+): boolean => {
+  return accessRequirements.filter(isARUnsupported).length > 0
+}
+
+const isARUnsupported = (accessRequirement: AccessRequirement) => {
+  return !SUPPORTED_ACCESS_REQUIREMENTS.has(accessRequirement.concreteType)
+}
+
+/**
+ * Represents a distinct screen in the wizard used to apply to a ManagedACTAccessRequirement
+ */
+enum RequestDataStep {
+  SHOW_ALL_ARS = 0,
+  UPDATE_RESEARCH_PROJECT = 1,
+  UPDATE_ACCESSORS_AND_FILES = 2,
+  PROMPT_CANCEL = 3,
+  PROMPT_LOGIN = 4,
+  COMPLETE = 5,
+}
+
+export type RequestDataStepCallbackArgs = {
+  step: RequestDataStep
+  managedACTAccessRequirement?: ManagedACTAccessRequirement
+  researchProjectId?: string
+  dataAccessRequest?: Request | Renewal
+}
+
+/**
+ * The AccessRequirementListV2 component renders a set of {@link AccessRequirement}s (either pre-defined, or those applied
+ * to an entity, based on provided props).
+ *
+ * The component shows the user the approval status of each AR and provides a workflow for accepting the terms of or
+ * creating a submission for any of the Access Requirements.
+ */
+export default function AccessRequirementListV2(
+  props: AccessRequirementListProps,
+) {
+  const {
+    entityId,
+    onHide,
+    accessRequirementFromProps,
+    renderAsModal,
+    numberOfFilesAffected,
+  } = props
+  const { accessToken } = useSynapseContext()
+  const isSignedIn = !!accessToken
+  const [requestDataStep, setRequestDataStep] = useState<RequestDataStep>(
+    RequestDataStep.SHOW_ALL_ARS,
+  )
+  const [managedACTAccessRequirement, setManagedACTAccessRequirement] =
+    useState<ManagedACTAccessRequirement>()
+  const [researchProjectId, setResearchProjectId] = useState<string>('')
+  const [dataAccessRequest, setDataAccessRequest] = useState<
+    Request | Renewal | undefined
+  >()
+
+  const entityHeaderProps: UseGetInfoFromIdsProps = {
+    ids: [entityId],
+    type: 'ENTITY_HEADER',
+  }
+
+  const entityInformation = useGetInfoFromIds<EntityHeader>(entityHeaderProps)
+
+  const { data: fetchedRequirements } = useGetAccessRequirementsForEntity(
+    entityId,
+    {
+      enabled: Boolean(!accessRequirementFromProps && entityId),
+    },
+  )
+
+  const accessRequirements = accessRequirementFromProps ?? fetchedRequirements
+
+  /**
+   * Set the initial ordering of the Access Requirements to show complete ARs first. The individual AR Item components
+   * retrieve and render status independently, so this data only determines ordering. For this reason, the query data
+   * should not be invalidated or updated as the user progresses through the workflow.
+   */
+  const { data: sortedAccessRequirementIds } =
+    useSortAccessRequirementIdsByCompletion(
```

Comment:
In v1, the analog to this function returned a sorted data object that contained the access requirements and their statuses.

Now, this is only concerned with ordering, so it just returns a sorted list of AR IDs. The individual access requirement components will show status.

This pattern lets us keep a stable ordering without holding on to stale status data as the user accepts terms and creates submissions.